open-feature/open-feature-operator
# fix: added namespace configuration for helm install using -n

## This PR


- adds a kustomize overlay to allow for helm configuration keys to be injected into the rendered yaml.
- updated helm configuration to restore expected functionality to `-n namespace-name`.
  
This pattern can also be used for number replacements following the example below:
```
containerPort: 0{{ .Values.numberValue }}
```
the `0` is required to pass linting when the strategic merge runs, but is dropped when the yaml is deployed
### Related Issues



### Notes


### Follow-up Tasks



### How to test

package the helm chart with the following command:
```
KUSTOMIZE_PATH=config/overlays/helm IMG=ghcr.io/open-feature/open-feature-operator:main make helm-package
```
install the helm chart with the following commands (only one deployment may exist at a time)
```
cd chart && helm upgrade ofo . --install -n test
```
kubectl create ns test
expected output: ofo deployed to `test` namespace, will fail if namespace does not already exist 
```
helm upgrade ofo . --install
```
expected output: ofo deployed to the default `open-feature-operator-system` namespace
